src/SiteInformation/FarmDates/EditDatesModal.js

```diff
@@ -0,0 +1,165 @@
+    const handleEditDatesModalClose = () => {
+        dispatch(setEditDatesModalOpen(false));
+    };
+
+    //handles submit
+    const handleEditDatesModalSubmit = () => {
+        //set data to be submitted to API
+        const data = {
+            "values": {
+                "cc_planting_date": datesDict.coverCropPlanting,
+                "cc_termination_date": datesDict.coverCropTermination,
+                "cash_crop_planting_date": datesDict.cashPlanting,
+            },
+            "conditions": {
+                "code": farmCode,
+            }
+        };
+        callAzureFunction( data, 'crowndb/farm_history', 'POST', getTokenSilently).then((res) => {
+            if (res.response && res.response.status === 201) {
+              dispatch(setEditDatesModalData({
+                    ...editDatesModalData,
+                    cover_planting: datesDict.coverCropPlanting,
+                    cover_termination: datesDict.coverCropTermination,
+                    cash_planting: datesDict.cashPlanting,
+              }));
+              dispatch(setFarmDatesReloadNeeded(true));
+              setSnackbarDataGlobal({
+                open: true,
+                text: 'Dates Edit Was Successful!',
+                severity: 'success',
+              });
+              handleEditDatesModalClose();
+            } else {
+                setSnackbarDataGlobal({
+                    open: true,
+                    text: 'Dates Edit Was Unsuccessful',
+                    severity: 'error',
+                });
+            }
+        });
+    };
+
+    return (
+       <Dialog 
+        open = {editDatesModalOpen} 
+        onClose={handleEditDatesModalClose}
+        >
+        <DialogTitle>Edit Dates</DialogTitle>
+        <DialogContent>
+            <Grid container spacing = {2}>
+                <Grid item xs={12}> 
+                    <DialogContentText>
+                        Please enter the new dates for Cover Crop Planting, Cover Crop Termination, and Cash Planting in the text fields below. 
+                    </DialogContentText>
+                </Grid>
+                <Grid item xs={4}> 
+                    <TextField 
+                        color = 'primary' 
+                        defaultValue={datesDict.coverCropPlanting}
+                        disabled = {datesDict.coverCropPlanting == null || datesDict.coverCropPlanting === 'undefined'}
+                        onChange={data => updateDate(0, data.target.value)} 
+                        type = 'date' 
+                        id = 'Cover Crop Planting' 
+                        label= 'Cover Crop Planting'
+                        variant = 'outlined'
+                        InputLabelProps={{ shrink: true }}
+                    />
+                </Grid>
+                <Grid item xs={4}> 
+                    <TextField 
+                        color = 'primary' 
+                        defaultValue={datesDict.coverCropTermination}
```

Comment:
same here, should be value. defaultValue is used as a placeholder not the actual value